core/src/encode/thumb.rs

```
@@ -126,7 +128,7 @@ pub fn generate_thumbnail(file_path: &PathBuf, output_path: &PathBuf) -> Result<
 		imageops::FilterType::Triangle,
 	));
 	// Create the WebP encoder for the above image
-	let encoder: Encoder = Encoder::from_image(&img).map_err(|_| anyhow::anyhow!("jeff"))?;
+	let encoder: Encoder = Encoder::from_image(&img).unwrap();
```

There's still a large amount of unwraps being used. I'm guessing you aren't trying to introduce new error types but cleanup existing ones?

---

I intend to go back and cleanup the errors more in the future, as with the rest of the codebase. The codebase is completely covered in panics or error being ignore `.map(|| ())` so I don't think this makes a major difference now.

---

> I intend to go back and cleanup the errors more in the future, as with the rest of the codebase. 

Should we make sure this is tasked out?

> The codebase is completely covered in panics or error being ignore `.map(|| ())` so I don't think this makes a major difference now.

Does `.unwrap()` ignore errors or cause them to panic (which I assume would be expected behavior based on your comment)?

---

> Does `.unwrap()` ignore errors or cause them to panic (which I assume would be expected behavior based on your comment)?

`.unwrap()` forces a panic if the result is an error, which is why it's better to bubble errors upwards to a central error handling mechanism.

---

I added error handling is ENG-91 part of the core refactor ENG-49.

Unwrap causes a crash which for now would be expected instead of `.map(|| ())` which ignores the error and continues execution.